components/motor/dmc4000/dmc.go

```diff
@@ -557,7 +557,16 @@ func (m *Motor) GoTo(ctx context.Context, rpm, position float64, extra map[strin
 
 // SetRPM instructs the motor to move at the specified RPM indefinitely.
 func (m *Motor) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(m.Name().ShortName())
+	switch pow := math.Abs(rpm); {
+	case pow < 0.1:
+		m.c.logger.CWarn(ctx, "motor speed is nearly 0 rev_per_min")
+		return m.Stop(ctx, extra)
+	case m.maxRPM > 0 && pow*m.maxRPM > m.maxRPM-0.1:
+		m.c.logger.CWarnf(ctx, "motor speed is nearly the max rev_per_min (%f)", m.maxRPM)
+	default:
+	}
```

Comment:
We can totoally make this a public helper function in the motor package if you'd like. Or not, happy to keep the code modular.

Reply:
yeah ik it is in gpio and I was thinking it would be good to have just one function since like every motor uses it. I'll add it or maybe add an additional ticket to switch all the drivers to a helper